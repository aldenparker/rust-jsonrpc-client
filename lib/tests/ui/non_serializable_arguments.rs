struct Number;

#[jsonrpc_client::api]
pub trait Math {
    async fn subtract(&self, subtrahend: Number, minuend: Number) -> u64;
}

fn main() {}
