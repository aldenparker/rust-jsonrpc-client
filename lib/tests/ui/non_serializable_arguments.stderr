error[E0277]: the trait bound `Number: Serialize` is not satisfied
 --> $DIR/non_serializable_arguments.rs:5:30
  |
5 |     async fn subtract(&self, subtrahend: Number, minuend: Number) -> u64;
  |                              ^^^^^^^^^^ the trait `Serialize` is not implemented for `Number`

error[E0277]: the trait bound `Number: Serialize` is not satisfied
 --> $DIR/non_serializable_arguments.rs:5:50
  |
5 |     async fn subtract(&self, subtrahend: Number, minuend: Number) -> u64;
  |                                                  ^^^^^^^ the trait `Serialize` is not implemented for `Number`
